README.md

```diff
@@ -119,7 +120,29 @@ that specific token. If omitted, the approval covers all tokens.
 
 The following APIs are not part of the fftokens standard, but are exposed under `/api/v1`:
 
-* `GET /receipt/:id` - Get receipt for a previous request
+- `GET /receipt/:id` - Get receipt for a previous request
+
+## Retry behaviour
+
+Most short-term outages should be handled by the blockchain connector. For example if the blockchain node returns `HTTP 429` due to rate limiting
+it is the blockchain connector's responsibility to use appropriate back-off retries to attempt to make the required blockchain call successfully.
+
+There are cases where the token connector may need to perform it's own back-off retry for a blockchain action. For example if the blockchain connector
+microservice has crashed and is in the process of restarting just as the token connector is trying to query an NFT token URI to enrich a token event, if
+the token connector doesn't perform a retry then the event will be returned without the token URI populated.
+
+The token connector has configurable retry behaviour for all blockchain related calls. By default the connector will perform up to 15 retries with a back-off
+interval between each one. The default first retry interval is 100ms and doubles up to a maximum of 10s per retry interval. Retries are only performed where
+the error returned from the REST call matches a configurable regular expression retry condition. The default retry condition is `._ECONN._` which ensures
+retries take place for common TCP errors such as `ECONNRESET` and `ECONNREFUSED`.
+
+Setting the retry condition to "" disables retries. The configurable retry settings are:
+
+- `RETRY_BACKOFF_FACTOR` (default `2`)
+- `RETRY_BACKOFF_LIMIT_MS` (default `10000`)
+- `RETRY_BACKOFF_INITIAL_MS` (default `100`)
+- `RETRY_CONDITION` (default `.*ECONN.*`)
+- `RETRY_MAX_ATTEMPTS` (default `15`)
```

Comment:
Can I configure an infinite max?

Reply:
The latest commit adds support for `RETRY_MAX_ATTEMPTS=-1` (infinite retries). I'm not 100% sure that my `for` loop logic is as clear as it could be - basically don't inc the counter when `retriesMax` is set to -1, so further comments on this are welcome.

Reply:
I think this is good, and the loop logic seems fine to me 👍